apps/newsletters-api/src/app/routes/currentStep.ts

```diff
@@ -47,27 +47,21 @@ export function registerCurrentStepRoute(app: FastifyInstance) {
 		'/v1/currentstep',
 		async (req, res): Promise<CurrentStepRouteResponse> => {
 			const body: CurrentStepRouteRequest = req.body;
-
-			if (body.newsletterId === undefined) {
-				return res
-					.status(400)
-					.send({ message: 'newsletter id is required', body: body });
-			}
-
 			let result: WizardStepData = { currentStepId: '' };
 			try {
-				result =
-					body.buttonId !== undefined
-						? await stateMachineButtonPressed(
-								body.buttonId,
-								{
-									currentStepId: body.stepId,
-									formData: body.formData,
-								},
-								newslettersWorkflowStepLayout,
-								storageInstance,
-						  )
-						: setupInitialState();
+				if (body.buttonId !== undefined) {
+					result = await stateMachineButtonPressed(
+						body.buttonId,
+						{
+							currentStepId: body.stepId,
+							formData: body.formData,
+						},
+						newslettersWorkflowStepLayout,
```

Comment:
As per your comment in the "newsletter-workflow" file, I think it would be clearer (and more scalable) is there were a separate `WizardLayout` object for the "launch newsletter" workflow. I think for that to work, the `CurrentStepRouteRequest` would need to include a "wizardId" field. The component now has the wizardId prop, so that should help.

Reply:
I have separated out the layouts